### PR TITLE
[Fix issue][#3511]lineplot of empty dataframe with hue in seaborn 0.13.0

### PR DESCRIPTION
**Pull request by Noko-Github**
Corresponds to the case where grouping_var has None object.

Related issue: https://github.com/mwaskom/seaborn/issues/3511

### CODE REVIEW

**Comment by mwaskom** [comment]
Thanks for taking a crack at this. Can you please add a test case that would fail with the current behavior (and pass with your fix)? Thanks!

File: seaborn/_base.py [hunk 1]
```diff
@@ -933,7 +933,8 @@ def iter_data(
 
             grouping_keys = []
             for var in grouping_vars:
-                grouping_keys.append(levels.get(var, []))
+                key = levels.get(var) if levels.get(var) is not None else []
+                grouping_keys.append(key)
```
Comment (mwaskom) [inline seaborn/_base.py:937]: Total nitpick, but the duplicate call to `levels.get` above looks cumbersome, maybe move the ternary down here? e.g.

```suggestion
                key = levels.get(var)
                grouping_keys.append([] if key is None else key)
```

**Review by mwaskom — CHANGES_REQUESTED** [review]
Thanks, a couple nitpicks but looks like the right approach.

**Comment by Noko-Github** [comment]
Thank you for your review!
I incorporated feedback.

**Comment by mwaskom** [comment]
Thanks @Noko-Github 

I think it's possible that this could also be fixed upstream by making the default `levels` value an empty list rather than `None`. I am not certain whether that would cause any other problems though and this is simple and seems to work so 👍

**Review by mwaskom — APPROVED** [review]

**Comment by mwaskom** [comment]
Btw if you make the PR message say "Closes #issue" or "Fixes #issue" it will close the issue automatically when merged.

**Pull request merged**